clifford/_layout.py

```diff
@@ -175,6 +179,201 @@ def construct_graded_mt(
+                op[scalar_index] += a
+                return op
+            return impl
+        elif isinstance(a, types.Array) and isinstance(b, types.abstract.Number):
+            def impl(a, b):
+                op = a.astype(np.float32)
+                op[scalar_index] += b
+                return op
+            return impl
+        else:
+            def impl(a, b):
+                return a + b
+            return impl
+
+    return ga_add
+
+
+def get_overload_sub(layout):
+    """
+    Returns an overloaded JITed function that works on
+    MultiVector value arrays
+    """
+    scalar_index = layout._basis_blade_order.bitmap_to_index[0]
+
+    def ga_sub(a, b):
+        # dummy function to overload
+        pass
+
+    @overload(ga_sub, inline='always')
+    def ol_ga_sub(a, b):
+        if isinstance(a, types.abstract.Number) and isinstance(b, types.Array):
+            def impl(a, b):
+                op = -b.astype(np.float32)
+                op[scalar_index] += a
+                return op
+            return impl
+        elif isinstance(a, types.Array) and isinstance(b, types.abstract.Number):
+            def impl(a, b):
+                op = a.astype(np.float32)
+                op[scalar_index] -= b
+                return op
+            return impl
+        else:
+            def impl(a, b):
+                return a - b
+            return impl
+
+    return ga_sub
+
+
+def get_overload_mul(layout):
+    """
+    Returns an overloaded JITed function that works on
+    MultiVector value arrays
+    """
+    def ga_mul(a, b):
+        # dummy function to overload
+        pass
+
+    gmt_func = layout.gmt_func
+    @overload(ga_mul, inline='always')
+    def ol_ga_mul(a, b):
+        if isinstance(a, types.Array) and isinstance(b, types.Array):
+            def impl(a, b):
+                return gmt_func(a, b)
+            return impl
+        else:
+            def impl(a, b):
+                return a*b
+            return impl
+
+    return ga_mul
+
+
+def get_overload_xor(layout):
+    """
+    Returns an overloaded JITed function that works on
+    MultiVector value arrays
+    """
+    def ga_xor(a, b):
+        # dummy function to overload
+        pass
+
+    as_ga = layout.as_ga_value_vector_func
+    omt_func = layout.omt_func
+    @overload(ga_xor, inline='always')
+    def ol_ga_xor(a, b):
+        if isinstance(a, types.Array) and isinstance(b, types.Array):
+            def impl(a, b):
+                return omt_func(a, b)
+            return impl
+        elif isinstance(a, types.Array) and isinstance(b, types.abstract.Number):
+            def impl(a, b):
+                return omt_func(a, as_ga(b))
+            return impl
+        elif isinstance(a, types.abstract.Number) and isinstance(b, types.Array):
+            def impl(a, b):
+                return omt_func(as_ga(a), b)
+            return impl
+        else:
+            def impl(a, b):
+                return a^b
+            return impl
```

Comment:
Tempting to make these shorter:
```suggestion
            return lambda a, b: omt_func(a, b)
        elif isinstance(a, types.Array) and isinstance(b, types.abstract.Number):
            return lambda a, b: omt_func(a, as_ga(b))
        elif isinstance(a, types.abstract.Number) and isinstance(b, types.Array):
            return lambda a, b: omt_func(as_ga(a), b)
        else:
            return lambda a, b: return a^b
```